brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/RuntimeEnvironmentModifier.java

```diff
@@ -49,41 +45,28 @@ public RuntimeEnvironmentModifier(ManagementContext mgmt, ToscaFacade<? extends
 
     @Override
     public void apply(EntitySpec<?> entitySpec, String nodeId, ToscaApplication toscaApplication) {
-        // TODO: Difference between artifacts on indexed node template and nodeTemplate?
         final Iterable<String> artifacts = getToscaFacade().getArtifacts(nodeId, toscaApplication);
         if (Iterables.isEmpty(artifacts)) {
             return;
         }
 
         final Map<String, String> filesToCopy = MutableMap.of();
-        final List<String> preInstallCommands = MutableList.of();
 
         for (String artifactId : artifacts) {
-
             Optional<Path> optionalResourcesRootPath = getToscaFacade().getArtifactPath(nodeId, toscaApplication, artifactId);
             if(!optionalResourcesRootPath.isPresent()) {
                 continue;
             }
 
-            // TODO: Use standard brooklyn directory rather than homedir.
-            final String destRoot = Os.mergePaths("~", "brooklyn-tosca-resources", artifactId);
-            preInstallCommands.add("mkdir -p " + destRoot);
-            // Artifact names may be referred to as environment variables in blueprint scripts.
-            entitySpec.configure(SoftwareProcess.SHELL_ENVIRONMENT.subKey(artifactId), destRoot);
-
-            final String tempRoot = Os.mergePaths("/tmp", artifactId);
-            preInstallCommands.add("mkdir -p " + tempRoot);
-
+            BrooklynDslDeferredSupplier deferredRunDir = BrooklynDslCommon.attributeWhenReady("run.dir");
+            entitySpec.configure(SoftwareProcess.SHELL_ENVIRONMENT.subKey(artifactId), deferredRunDir);
 
             // Copy all files in resource.
             try {
                 Files.walkFileTree(optionalResourcesRootPath.get(), new SimpleFileVisitor<Path>() {
                     @Override
                     public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                        String tempDest = Os.mergePaths(tempRoot, file.getFileName().toString());
-                        String finalDest = Os.mergePaths(destRoot, file.getFileName().toString());
-                        filesToCopy.put(file.toAbsolutePath().toString(), tempDest);
-                        preInstallCommands.add(String.format("mv %s %s", tempDest, finalDest));
+                        filesToCopy.put(file.toAbsolutePath().toString(), file.getFileName().toString());
```

Comment:
Doesn't this risk overwriting files? e.g. if a project had resources `config/foo` and `config2/foo`. If possible the approach should be to copy the files with their directory structure intact.